### PR TITLE
IOS15 have failed on the old way, USES the UINavigationBarAppearance …

### PR DESCRIPTION
**Pull request by SunShineLOL**
The old method on iOS15 is no longer valid, use UINavigationBarAppearance to set UINavigationBar

### CODE REVIEW

**Review by emrcftci — APPROVED** [review]

**Comment by emrcftci** [comment]
could we declare the extension as an open? lastly we've needed to override `makeTransparent` method but couldn't.

File: Sources/SwifterSwift/UIKit/UINavigationBarExtensions.swift [hunk 1]
```diff
@@ -15,20 +15,35 @@ public extension UINavigationBar {
...
         attrs[.foregroundColor] = color
-        titleTextAttributes = attrs
+        if #available(iOS 13.0, *) {
+            let standardApp = standardAppearance
+            standardApp.titleTextAttributes = attrs
+            standardAppearance = standardApp
+            if let scrollApp = scrollEdgeAppearance {
+                scrollApp.titleTextAttributes = attrs
+                scrollEdgeAppearance = scrollApp
+            }
```
Comment (guykogus) [inline Sources/SwifterSwift/UIKit/UINavigationBarExtensions.swift:25]:
```suggestion
            standardAppearance.titleTextAttributes = attrs
            scrollEdgeAppearance?.titleTextAttributes = attrs
```

File: Sources/SwifterSwift/UIKit/UINavigationBarExtensions.swift [hunk 2]
```diff
@@ -15,20 +15,35 @@ public extension UINavigationBar {
...
+        }else{
```
Comment (guykogus) [inline Sources/SwifterSwift/UIKit/UINavigationBarExtensions.swift:26]:
```suggestion
        } else {
```

File: Sources/SwifterSwift/UIKit/UINavigationBarExtensions.swift [hunk 3]
```diff
@@ -38,11 +53,24 @@ public extension UINavigationBar {
...
+            let standardApp = standardAppearance
+            standardApp.backgroundColor = background
+            standardApp.backgroundImage = UIImage()
+            standardApp.titleTextAttributes = [.foregroundColor: text]
+            standardAppearance = standardApp
+            if let scrollApp = scrollEdgeAppearance {
+                scrollApp.backgroundColor = background
+                scrollApp.backgroundImage = UIImage()
+                scrollApp.titleTextAttributes = [.foregroundColor: text]
+            }
```
Comment (guykogus) [inline Sources/SwifterSwift/UIKit/UINavigationBarExtensions.swift:67]:
```suggestion
            standardAppearance.backgroundColor = background
            standardAppearance.backgroundImage = UIImage()
            standardAppearance.titleTextAttributes = [.foregroundColor: text]

            scrollEdgeAppearance?.backgroundColor = background
            scrollEdgeAppearance?.backgroundImage = UIImage()
            scrollEdgeAppearance?.titleTextAttributes = [.foregroundColor: text]
```

File: Sources/SwifterSwift/UIKit/UINavigationBarExtensions.swift [hunk 4]
```diff
@@ -38,11 +53,24 @@ public extension UINavigationBar {
...
+        } else{
```
Comment (guykogus) [inline Sources/SwifterSwift/UIKit/UINavigationBarExtensions.swift:68]:
```suggestion
        } else {
```

File: Sources/SwifterSwift/UIKit/UINavigationBarExtensions.swift [hunk 5]
```diff
@@ -15,20 +15,35 @@ public extension UINavigationBar {
...
-        backgroundColor = .clear
-        barTintColor = .clear
-        setBackgroundImage(UIImage(), for: .default)
-        tintColor = tint
-        titleTextAttributes = [.foregroundColor: tint]
-        shadowImage = UIImage()
+        if #available(iOS 13.0, *) {
+            standardAppearance.configureWithTransparentBackground()
+            scrollEdgeAppearance?.configureWithTransparentBackground()
+        }else{
```
Comment (guykogus) [inline Sources/SwifterSwift/UIKit/UINavigationBarExtensions.swift:39]:
```suggestion
        } else {
```

File: Sources/SwifterSwift/UIKit/UINavigationBarExtensions.swift [hunk 6]
```diff
@@ -15,20 +15,35 @@ public extension UINavigationBar {
         var attrs = [NSAttributedString.Key: Any]()
         attrs[.font] = font
         attrs[.foregroundColor] = color
-        titleTextAttributes = attrs
+        if #available(iOS 13.0, *) {
```
Comment (guykogus) [inline Sources/SwifterSwift/UIKit/UINavigationBarExtensions.swift:18]: You might need to add checks for `tvOS 13.0` as well

**Review by guykogus — CHANGES_REQUESTED** [review]
Thanks for the update. You need to add a CHANGELOG entry and maybe some tests, if possible (although I'm not sure how).

**Pull request closed without merging**